src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -126,24 +128,27 @@ trait RegularSync {
         //we have same chain prefix
         if (parent.hash == headers.head.parentHash) {
 
-          val currentBranchTotalDifficulty = headersQueue.map(_.number)
+          val oldBlocks: Seq[Option[(BlockBody, BigInt)]] = headersQueue.map(_.number)
             .map(blockNumber => blockchain.getBlockByNumber(blockNumber))
-            .collect{
-              case Some(b) => b.header.difficulty
-              case None => BigInt(0)
-            }.sum
+            .map{_.map{b => (b.body, b.header.difficulty)}}
+
+          val currentBranchTotalDifficulty: BigInt = oldBlocks.collect {
+            case Some((_, difficulty)) => difficulty
+          }.sum
 
           val newBranchTotalDifficulty = headersQueue.map(_.difficulty).sum
 
           if (currentBranchTotalDifficulty < newBranchTotalDifficulty) {
+            val transactionsToAdd = oldBlocks.collect{case Some((blockBody,_)) => blockBody.transactionList}.flatten
+            actors.pendingTransactionsManager ! PendingTransactionsManager.AddTransactions(transactionsToAdd.toList)
```

Comment:
yes we do i added tests in https://github.com/input-output-hk/etc-client/pull/196 I can merge them back here
also I was thinking about merging #193 and #196 together to have one PR